database/sql/V072__auction_solution_orders.sql

```diff
@@ -0,0 +1,36 @@
+-- Table to store all proposed solutions for an auction, received from solvers during competition time.
+-- A single auction can have multiple solutions, and each solution can contain multiple orders.
+-- Order is not allowed to be part of more than one solution within the same auction.
+
+-- `solver` + `solution_id` are unique within an auction.
+
+-- `deadline` is used to determine if the order is part of a winning solution.
+
+CREATE TABLE auction_solution_orders (
+   auction_id bigint NOT NULL,
+   solver bytea NOT NULL,
+   solution_id string NOT NULL,
+   order_uid bytea NOT NULL,
+   -- The block number until which the order should be settled.
+   -- Not NULL for winning orders.
+   deadline bigint,
```

Comment:
The `deadline` is a property of the `auction` and not the order.

Reply:
Assuming there is no bug we should be able to reconstruct who (should have) won with all the data we have, no?

The criteria might change even on each restart of the autopilot (for example, if we enable/disable multiple winners feature several times). In this case we would need to know who was supposed to settle solutions.

Reply:
I've refactored the tables before you posted a comment. Can you check if it's more acceptable now?

Reply:
> The criteria might change even on each restart of the autopilot (for example, if we enable/disable multiple winners feature several times). In this case we would need to know who was supposed to settle solutions.

Alternative to this is to also save the information "which type of competition" was executed for each auction. With this, we would have an input and would be able to determine the winners so "is_winner" would not be needed.